core/services/scan.go

```diff
@@ -148,32 +149,36 @@ func (s *ScanService) ScanAP(ctx context.Context) error {
 		helpers.String("namespace", namespace),
 		helpers.String("jobID", workload.JobID))
 
-	scans, err := s.relevancyProvider.GetContainerRelevancyScans(ctx, namespace, name)
+	scans, err := s.relevancyProvider.GetContainerRelevancyScans(mainCtx, namespace, name)
 	if err != nil {
 		return fmt.Errorf("error getting container relevancy scans: %w", err)
 	}
 
 	for _, scan := range scans {
 		slug, err := names.ImageInfoToSlug(scan.ImageTag, scan.ImageID)
 		if err != nil {
-			logger.L().Ctx(ctx).Warning("error getting image slug", helpers.Error(err),
+			logger.L().Ctx(mainCtx).Warning("error getting image slug", helpers.Error(err),
 				helpers.String("imageTag", scan.ImageTag),
 				helpers.String("imageID", scan.ImageID))
 			continue // we need the slug
 		}
 
+		instanceIDSlug, _ := scan.InstanceID.GetSlug(false)
+
 		// create a workload inside a new context
-		ctx = context.WithValue(ctx, domain.WorkloadKey{}, domain.ScanCommand{
+		ctx := enrichContext(mainCtx, domain.ScanCommand{
 			JobID:              uuid.NewString(),
 			ImageSlug:          slug,
 			ContainerName:      scan.ContainerName,
-			ImageHash:          scan.ImageID,
+			ImageHash:          v1.NormalizeImageID(scan.ImageID, scan.ImageTag),
 			ImageTagNormalized: scan.ImageTag,
-			InstanceID:         scan.InstanceIDString,
+			InstanceID:         instanceIDSlug,
```

Comment:
Before using ScanAP API, the operator used to call ScanCVE API with `InstanceId` with the slug. (see https://github.com/kubescape/operator/blob/v0.2.49/mainhandler/vulnscan.go#L517)

In the new flow, the operator calls `ScanAP`, with empty `InstanceID` - which means, the logic has moved to kubevuln.
In addition, `InstanceIDString` (contains `/`) which is not aligned with the logic of `ValidateContainerScanID` - when using `InstanceIDString` it will cause a generation of new scan ID each time. this is why we are using instanceIDSlug.